packages/flutter/lib/src/rendering/object.dart

```diff
@@ -756,18 +756,65 @@ class _ForkingSemanticsFragment extends _SemanticsFragment {
   }
 }
 
+/// The pipeline owner drives the rendering pipeline.
+///
+/// Every time the framework needs to produce a frame, the pipeline owner
+/// flushes any pending updates in the render tree.
+///
+/// 1. Any render objects that need layout get their layout information updated
+///    by [flushLayout]. During this phase, the size and position of each render
+///    object is calculated. Render objects might dirty their painting or
+///    compositing state during this phase.
+/// 2. Any render objects that have dirty compositing bits get cleaned by
+///    [flushCompositingBits]. During this phase, each render object learns
+///    whether any of its children require compositing. This information is used
+///    during the painting phase when selecting how to implement visual effects
+///    such as clipping. If a render object has a composited child, its needs
+///    to use a [Layer] to create the clip in order for the clip to apply to
+///    the composited child (which will be painted into its own [Layer]).
+/// 3. Any render objects that need to paint will be visited by [flushPaint].
+///    During this phase, render objects get a chance to record painting
+///    commands into [PictureLayer]s and construct other composited [Layer]s.
+/// 4. Finally, if [SemanticsNode.hasListeners] is true, [flushSemantics] will
+///    compile the semantics for the render objects. This semantic information
+///    is used by assistive technology to improve the accessibility of the
+///    render tree.
```

Comment:
We should probably talk about how you can have disconnected pipeline owners too.